src/net/client/clock.rs

```diff
@@ -0,0 +1,127 @@
+//! A time interface that can be replaced by a fake time implementation
```

Comment:
This is nice, but not needed.

If `Instant` in `net/client/cache.rs` is replaced by `tokio::time::Instant` and in `do_client()` when `TIME_PASSES` one calls `tokio::time::advance()` instead of `clock.adjust_time()` then time is normal unless one annotates a test with `#[tokio::test(start_paused = true)]` instead of `#[tokio::test]`. Then only the call to `advance()` causes time to advance. If needed one can also start with time running and only "pause" it (so that it can be advanced) at certain points.

Reply:
I'll resolve this because it is now in a separate PR.